test/unit/pdf_find_controller_spec.js

```diff
@@ -1137,4 +1137,24 @@ describe("pdf_find_controller", function () {
       });
     });
   });
+
+  it("performs a search in a text with compound word on two lines", async function () {
+    const { eventBus, pdfFindController } =
+      await initPdfFindController("issue18693.pdf");
+
+    await testSearch({
+      eventBus,
+      pdfFindController,
+      state: {
+        query: "hel-Lo",
+      },
+      matchesPerPage: [1],
+      selectedMatch: {
+        pageIndex: 0,
+        matchIndex: 0,
+      },
+      pageMatches: [[6]],
+      pageMatchesLength: [[7]],
+    });
+  });
```

Comment:
Nit: Can we please move this test *before* the `"custom matcher"` block instead, i.e. place it here?
https://github.com/mozilla/pdf.js/blob/0ac7f294cdb84bb0e4a346e890f567a1574f531c/test/unit/pdf_find_controller_spec.js#L1064